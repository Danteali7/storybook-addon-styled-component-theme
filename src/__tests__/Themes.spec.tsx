import * as React from "react";
import {stub} from "sinon";
import {Themes} from "../Themes";
import {create} from "react-test-renderer";

describe("Themes spec", () => {
    it("should render proper", () => {
        const channel = {
            on: stub(),
            emit: stub(),
            removeListener: stub(),
        };

        const component = create(<Themes api={null} channel={channel} active={true} />);
        expect(component.toJSON()).toMatchSnapshot();
        expect(channel.on.calledOnce).toBeTruthy();
        expect(channel.emit.notCalled).toBeTruthy();

        component.unmount();
        expect(channel.removeListener.calledOnce).toBeTruthy();
    });
});
