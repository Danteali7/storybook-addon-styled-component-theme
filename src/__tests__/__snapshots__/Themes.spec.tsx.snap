// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themes spec should render proper 1`] = `
<div
  class="sc-bdfBwQ cccArU"
>
  <div
    class="sc-gsTCUz hwEMSL"
  />
  <div
    class="sc-dlfnbm jrWvId"
  >
    |
  </div>
</div>
`;
